compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/Check.hs

```diff
@@ -656,13 +656,19 @@ checkDefValue (DefValue _loc (_, typ) _noParties (IsTest isTest) expr) = do
       _ -> throwWithContext (EExpectedScenarioType typ)
 
 checkTemplateChoice :: MonadGamma m => Qualified TypeConName -> TemplateChoice -> m ()
-checkTemplateChoice tpl (TemplateChoice _loc _ _ actors selfBinder (param, paramType) retType upd) = do
+checkTemplateChoice tpl (TemplateChoice _loc _ _ controllers observers selfBinder (param, paramType) retType upd) = do
   checkType paramType KStar
   checkType retType KStar
-  introExprVar param paramType $ checkExpr actors (TList TParty)
+  introExprVar param paramType $ checkExpr controllers (TList TParty)
+  introExprVar param paramType $ checkOptionalChoiceObservers observers
```

Comment:
Rename `observers` above to `mbObservers` and
```suggestion
  introExprVar param paramType $ do
    checkExpr controllers (TList TParty)
    whenJust mbObservers $ \observers -> do
      -- check version is recent enough here
      checkExpr observers (TList TParty)
```
and drop `checkOptionalChoiceObservers` below.